### PR TITLE
Private API attachments

### PR DESCRIPTION
**Pull request by mrsylerpowers**
Currently only added for Mac OS 10 because that's all the OS's I have access to
Params for each event 
## Event `new-transfer`
The new transfer event allows for files to be uploaded and registered with the file transfer daemon. Files registered with the file transfer daemon are ready to be attached to new messages.
**Warning** Files that sit here for to long can be purged by mac os and will not be available for new messages
### Expects 
Param Required*:  `filePath` the file path local to the computer
> Must be in a location "Messages.app" can access. Most locations are not accessible by "Messages.app" one safe location is `~/Library/Messages`
### Returns
Param: `identifier` Which is the file transfer guid that can be used to send a later message
## Event `send-message`
Send Message has been updated to accept a new param that allows for multipart messages & attachments with send effects
### Expects 
Param *Required for file sending*: `fileTransferGUIDs` This  should be a array of the fileTransferGuids that are about to be sent
Param *Required for file sending*: `attributedBody` must be sent with proper attributes and string detailing the message. Attachments placed should use the object-replacment-char `\ufffc`  with proper ranges
> If not sent with proper attributed body correctly attachments will not show in UI 

Example of the attributes that need to be present on the object replacement char
```                      
 "attributedBody": [
          {
               "string": "￼￼",
               "runs": [
                    {
                         "range": [
                              0,
                              1
                         ],
                         "attributes": {
                              "__kIMMessagePartAttributeName": 0,
                              "__kIMFileTransferGUIDAttributeName": "13772C3A-E9CD-481E-AC98-C4B317603060",
                              "__kIMFileNameAttributeName": "something.jpg"
                         }
                    },
                    {
                         "range": [
                              1,
                              1
                         ],
                         "attributes": {
                              "__kIMMessagePartAttributeName": 1,
                              "__kIMFileTransferGUIDAttributeName": "DB44480B-DF4C-4011-81B8-1F44843C6EE0",
                              "__kIMFileNameAttributeName": "something2.jpg"
                         }
                    }
               ]
          }
     ]
```
## Event `send-attachment`
The sent attachment event allows for transfers and sending message in one go. There is no message body allowed just simple attachment sending
### Expects 
Param Required*:  `filePath` the file path local to the computer
> Must be in a location "Messages.app" can access. Most locations are not accessible by "Messages.app" one safe location is `~/Library/Messages`
Param Required*: `chatGuid` the chat the attachment should be sent to.
### Returns
Param: `identifier` Which is the message guid that was just sent

### CODE REVIEW

File: MacOS-10/BlueBubblesHelper/BlueBubblesHelper.m [hunk 1]
```diff
@@ -354,6 +422,23 @@ -(void) handleMessage: (NetworkController*)controller  message:(NSString *)messa
...
+        if (chat != nil && filePath !=nil) {
+
+            NSString *effectId = nil;
+            if (data[@"effectId"] != [NSNull null] && [data[@"effectId"] length] != 0) {
+                effectId = data[@"effectId"];
+            }
+
+            [BlueBubblesHelper sendFileTransferToChat:chat filePath:filePath effectId:effectId transaction:transaction];
+
+            if (transaction != nil) {
```
Comment (mrsylerpowers) [inline MacOS-10/BlueBubblesHelper/BlueBubblesHelper.m:437]: Should be removed

**Review by mrsylerpowers — COMMENTED** [review]
-1

**Comment by mrsylerpowers** [comment]
Just now added ability to send multipart attachments + effects with attachments

**Review by tneotia — APPROVED** [review]
LGTM. Simply amazing!

**Pull request merged**